staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -93,7 +93,6 @@ func NewWaitFlags(restClientGetter genericclioptions.RESTClientGetter, streams g
 			WithFieldSelector("").
 			WithAll(false).
 			WithAllNamespaces(false).
-			WithLocal(false).
```

Comment:
The change although good, is unfortunately breaking the API, before this PR `--local` was supported, after it's not there. So in your example you'll actually get:
```
$ kubectl wait --for=condition=ready --local -f pod.yaml
Error: unknown flag: --local
See 'kubectl wait --help' for usage.
```
Even though this is experimental command, I'd prefer not changing the API, but fix it appropriately. This means wiring the value of `--local` flag properly. Actually this is a bigger effort across all of the commands.

Reply:
> The change although good, is unfortunately breaking the API

@zhouya0 what was the behavior of that flag before this change? If it was completely broken (a.k.a failure and/or panic), then I would hardly consider it a breaking change.

Reply:
Yes, before it was a panic. But seems @soltysh will refactor work with `--local` and `--dry-run=client` and maybe it will be fixed after that. :)

Reply:
What do you expect to get from a "client-side local wait" though?

Reply:
Yes! I agree with you... User really should not use `kubectl wait --local`. I just don't know how to do with this PR now...